src/dstack/_internal/server/background/tasks/process_instances.py

```diff
@@ -557,6 +589,18 @@ async def _create_instance(session: AsyncSession, instance: InstanceModel) -> No
         instance.started_at = get_current_datetime()
         instance.last_retry_at = get_current_datetime()
 
+        if shared_info is not None:
+            if blocks == "auto":
+                blocks = len(instance_offer.instance.resources.gpus)
+                if blocks > 1:
+                    shared_info.total_blocks = blocks
+                else:
+                    shared_info = None
+            if shared_info is not None:
+                instance.shared_info = shared_info.json()
+            else:
+                instance.shared_info = None
```

Comment:
The `shared_info` updating is convoluted. I believe it can be created with `total_blocks: "auto"` and then it's overridden. I don't see much value in that. I'd keep user's configured `blocks` as a static property somewhere and wound't touch it. `total_blocks` can then be resolved when instance resources are known. No need to allow for `auto`.

Reply:
Replaced with NULL -> calcualted auto blocks in https://github.com/dstackai/dstack/pull/2253/commits/8b555f7a6ad65afbed244e3d1e76359bdbbcbbf9

NULL as "info is not yet available" seems to fit there